packages/tooltip/src/types.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { HTMLProps, RefObject } from 'react';
+
+export interface IUseTooltipProps<T = HTMLElement> {
+  /** Sets delay to the opening and closing of the tooltip */
+  delayMilliseconds?: number;
+  /** Specifies the tooltip ID */
+  id?: string;
+  /** Displays the tooltip on initial render */
+  isVisible?: boolean;
+  /** Provides ref access to the underlying trigger element */
+  triggerRef: RefObject<T>;
```

Comment:
Since requiring `triggerRef` will break existing implementations, especially for users not leveraging `aria-expanded`, I’d suggest keeping it optional. That way, we offer enhanced behavior without forcing an immediate change.

```suggestion
  /** If provided, the tooltip will automatically close when the trigger has `aria-expanded="true"` */
  triggerRef?: RefObject<T>;
```

Reply:
> this will not result in a breaking change to `react-components`

oh, oops, my latest PR description edit didn't make it in 😜